### DISCUSSION TITLE
Both checkbox must be checked to apply CPU limit PL1+PL2 ?

### DISCUSSION THREAD

**Discussion by Kikimahe** [seerge/g-helper#1759] [Q&A]
Hi,

First, sorry if I didn't understand or if this has already been explained (I couldn't find it in the search).

When I apply a CPU limit, I need to check apply limit **AND** necessarily apply the curve.  Otherwise, the limit is not applied (!?)
 * Is this normal operation ?
* And if so, what is the “apply limit” checkbox for?

[log.txt](https://github.com/seerge/g-helper/files/13705861/log.txt)

16h19 : Limit the processor to 30 W with apply limit only  ---> Limite isn't enable 
![01](https://github.com/seerge/g-helper/assets/142881210/a79a1209-5514-46f9-9d07-5bb5ccd49d47)

16h28 : Limit the CPU 30W with apply limit **AND apply curve** ---> OK Limit enable
![02](https://github.com/seerge/g-helper/assets/142881210/fa945f3e-0c92-4818-a9a2-821a356ac25f)

**Comment by seerge** [comment]
@Kikimahe hello, most of the models allow setting them separately. But some (for example G14 2023)  require both to be applied at the same time (as in Armoury you can't set only fans or only power limits in custom mode anyway)

If you know for sure, that your device is affected - I can add a "fix" where G-Helper would just silently apply current fan curve next to power limits on your device

**Comment by Kikimahe** [comment]
Hi and thanks Seerge.

No, don't make a fix just for me. Now, with your explanations, I understand better how it works. 
(I haven't used Armory Crate enough, to know it worked like that).

Some new Ghelper users will be perhaps surprised that CPU limit doesn't works, even if "apply limits" is checked (like I was). :-)
